client/boot/index.js

```diff
@@ -361,6 +362,20 @@ function reduxStoreReady( reduxStore ) {
 		require( 'lib/rubberband-scroll-disable' )( document.body );
 	}
 
+	/*
+	 * Layouts with differing React mount-points will not reconcile correctly,
+	 * so remove an existing single-tree layout by re-rendering if necessary.
+	 *
+	 * TODO (@seear): React 15's new reconciliation algo may make this unnecessary
+	 */
+	page( '*', function( context, next ) {
+		if ( ! isIsomorphicRoute( context.path ) && previousLayoutIsSingleTree() ) {
```

Comment:
Isn't the `! isIsomorphicRoute()` check obsolete, too? I think the `previousLayoutIsSingleTree` cases are an actual subset of the "isomorphic" (actually `router`-instead-of-`page`) ones...